xclim/sdba/properties.py

```diff
@@ -947,6 +1004,147 @@ def _spatial_correlogram(da: xr.DataArray, *, dims=None, bins=100, group="time")
 )
 
 
+def _decorrelation_length(
+    da: xr.DataArray, *, radius=300, thresh=0.50, dims=None, bins=100, group="time"
+):
+    """Decorrelation length.
+
+    Distance from a grid cell where the correlation with its neighbours goes below the threshold.
+    A correlogram is calculated for each grid cell following the method from ``xclim.sdba.properties.spatial_correlogram``.
+    Then, we find the first bin closest to the correlation threshold.
+
+    Parameters
+    ----------
+    da: xr.DataArray
+      Data.
+    radius: int
+        Radius (in km) defining the region where correlations will be calculate between a point and its neighbours.
+    thresh: float
+      Threshold correlation defining decorrelation.
+       The decorrelation length is defined as the center of the distance bin that has a correlation closest to this threshold.
+    dims: sequence of strings
+      Name of the spatial dimensions. Once these are stacked, the longitude and latitude coordinates must be 1D.
+    bins:
+      Same as argument `bins` from :py:meth:`scipy.stats.binned_statistic`.
+      If given as a scalar, the equal-width bin limits from 0 to radius are generated here
+      (instead of letting scipy do it) to improve performance.
+    group: str
+      Useless for now.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+      Decorrelation length.
+
+    Notes
+    -----
+    Calculating this property requires a lot of memory. It will not work with large datasets.
+    """
+    # save attrs
+    coords_attrs = {}
+    for c in da.coords:
+        coords_attrs[c] = da[c].attrs
```

Comment:
Could `xclim.sdba.utils.copy_all_attrs` be used for the same goal?

Reply:
yes `copy_all_attrs(out,da)` would do the trick here.

but I find the doctring of `copy_all_attrs` a bit confusing... | read "Copy all attributes of ds to ref," as take attrs from ds and put them in ref, but the fonction does it the other way around.

Reply:
Indeed! I can fix this in this PR.